android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -258,7 +258,10 @@ public void shareIntent(final ReadableMap options, final Callback callback) {
         }
         if (options.hasKey("uri")) {
             filePath = options.getString("uri");
-            uri = Uri.parse(filePath);
+
+            // http://stackoverflow.com/questions/33274470/from-uri-to-path-and-back-to-uri
+            File file = new File(filePath);
+            uri = Uri.fromFile(file);
```

Comment:
You should provide `filePath` with `file://`, not wrap by `File`. If you provide `filePath` with `content://`, this fix will fail in this case